### PR TITLE
blobwars: bump to 2.00

### PR DESCRIPTION
**Pull request by R4H33M**
Tested on x86_64.

### CODE REVIEW

File: games-puzzle/blobwars/blobwars-2.00.recipe [hunk 1]
```diff
@@ -9,10 +9,10 @@ evil alien leader, Galdov."
 HOMEPAGE="https://www.parallelrealities.co.uk/p/blob-wars-metal-blob-solid.html"
 COPYRIGHT="2004, 2005 Milan Babuskov"
 LICENSE="GNU GPL v2"
-REVISION="3"
-SOURCE_URI="http://sourceforge.net/projects/blobwars/files/blobwars-1.19.tar.gz"
-CHECKSUM_SHA256="f9bafe59ead9e370ceaba4aae3d280241bf05fa06c8ae006eec0322c2fe9e2ed"
-PATCHES="blobwars-1.19.patch"
+REVISION="4"
```
Comment (korli) [inline games-puzzle/blobwars/blobwars-2.00.recipe]:
```suggestion
REVISION="1"
```

**Comment by korli** [comment]
Please reset the revision. Otherwise looks good.

**Comment by Begasus** [comment]
Worked on this 4 years ago already, checking if it "runs" now :)

**Comment by korli** [comment]
> Worked on this 4 years ago already, checking if it "runs" now :)

Maybe it doesn't build on gcc2 anymore.

**Comment by Begasus** [comment]
Missing cmd:pkg_config

**Comment by Begasus** [comment]
> > Worked on this 4 years ago already, checking if it "runs" now :)
> 
> Maybe it doesn't build on gcc2 anymore.

Sill builds, as it did back then, but couldn't launch it: "The game is starting but doesn't fine a GL renderer" (comment on my work I added in 2016) :)

**Comment by R4H33M** [comment]
> Sill builds, as it did back then, but couldn't launch it: "The game is starting but doesn't fine a GL renderer" (comment on my work I added in 2016) :)

Should I remove gcc2 as a build target or attempt to fix it? (if it is possible, considering that you tried it 4 years ago. :))

File: games-puzzle/blobwars/patches/blobwars-2.00.patchset [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ 
+ OBJS += CAudio.o
+-- 
+2.19.0
+
+
+From fe851b0071135e4b704839bb16b6685479d0ad7b Mon Sep 17 00:00:00 2001
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Wed, 15 Jan 2020 12:10:23 +0000
+Subject: for tracker and deskbar
```
Comment (Begasus) [inline games-puzzle/blobwars/patches/blobwars-2.00.patchset]: Needs another patch to launch from Deskbar

**Comment by Begasus** [comment]
Nice, it launches now (from Tracker and Terminal, not from Deskbar)
Game is slow compared to the previous version (which uses SDL instead of SDL2)
Game doesn't close on exit, need to kill it with ProcessControler

**Comment by R4H33M** [comment]
I'm guessing these issues are gcc2 specific because it launches from deskbar and closes as it should from my testing on x64. What are your suggestions for fixing them?

**Comment by Begasus** [comment]
> I'm guessing these issues are gcc2 specific because it launches from deskbar and closes as it should from my testing on x64. What are your suggestions for fixing them?

Leave x86_gcc2 enabled, it builds ok and runs, so no need to disable it
For the Deskbar launch, not sure it's x86_gc2 specific, but there already some fixes at haikuports for it :)

**Comment by Begasus** [comment]
No harm in sharing it here, my work back then, you had a bit more in this PR :)
https://github.com/Begasus/haikuports/commit/f533fcf81f731ec586203b90b0bc29f68bd4c922

**Comment by R4H33M** [comment]
I've pushed a potential fix for the "launch from deskbar" gcc2 issue. Could you test it?

File: games-puzzle/blobwars/patches/blobwars-2.00.patchset [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ static inline void strlcpy(char *dest, const char *src, size_t n) { strncpy(dest, src, n); dest[n - 1] = 0; }
+ #endif
+-- 
+2.19.0
+
+
+From be609d5c68a6509b3507be3ddfc1ac4167678fc3 Mon Sep 17 00:00:00 2001
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Wed, 15 Jan 2020 18:21:04 +0000
+Subject: Allows launch from tracker and deskbar
```
Comment (Begasus) [inline games-puzzle/blobwars/patches/blobwars-2.00.patchset]: Shouldn't affect the patchset, but could you use "Tracker" and "Deskbar" here?

**Comment by Begasus** [comment]
> I've pushed a potential fix for the "launch from deskbar" gcc2 issue. Could you test it?

Got another (pretty heavy) build running atm, will check later, thanks!

File: games-puzzle/blobwars/patches/blobwars-2.00.patchset (hunk 3 above)
Comment (R4H33M) [reply games-puzzle/blobwars/patches/blobwars-2.00.patchset]: Done!

**Comment by Begasus** [comment]
OK, launches from Deskbar now, thanks!
I'm still having an issue with blobwars not being closed down properly, it remains silent (closes) but process is still running after closing it (gcc2h)

**Comment by Begasus** [comment]
@R4H33M Maybe keep the previous recipe and patchset, and mark this one as "?" for the architectures?

**Comment by R4H33M** [comment]
New commit. Was this what you meant?

**Comment by Begasus** [comment]
Thanks, how does it run on x86_64? Last time I checked on 32bit it was lagging (slow)

**Comment by R4H33M** [comment]
I decided to give it a test again now and it is really strange as I remember it working very nicely, but it was very laggy and slow.
It could be because I tested it before on my desktop computer (more capable hardware)

I hope we can still go ahead and merge, though.

File: games-puzzle/blobwars/blobwars-2.00.recipe [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+evil alien leader, Galdov."
+HOMEPAGE="https://www.parallelrealities.co.uk/p/blob-wars-metal-blob-solid.html"
+COPYRIGHT="2004, 2005 Milan Babuskov"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="http://sourceforge.net/projects/blobwars/files/blobwars-$portVersion.tar.gz"
+CHECKSUM_SHA256="c406279f6cdf2aed3c6edb8d8be16efeda0217494acd525f39ee2bd3e77e4a99"
+PATCHES="blobwars-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
```
Comment (Begasus) [inline games-puzzle/blobwars/blobwars-2.00.recipe]: Maybe disable x86_64 also, as it seems it's lagging there too?
```diff
- ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+ ARCHITECTURES="?x86_gcc2 ?x86 ?x86_64"
```

**Comment by Begasus** [comment]
@korli any thought on this? It builds OK, and runs, bug is lagging (don't think it's a bug on their side, seen some strange things when using SDL2 already)

**Comment by R4H33M** [comment]
Any updates on this? Are we good to merge?

**Comment by Begasus** [comment]
> Any updates on this? Are we good to merge?

If you could disable the build for the new version I'm fine with merging :)
https://github.com/haikuports/haikuports/pull/4587#pullrequestreview-347911952

**Comment by Begasus** [comment]
> > Any updates on this? Are we good to merge?
> 
> If you could disable the build for the new version I'm fine with merging :)
> [#4587 (review)](https://github.com/haikuports/haikuports/pull/4587#pullrequestreview-347911952)

@R4H33M could you take care of the review so we can merge this, tried to push a change there but seems I can't from my side :)

**Comment by R4H33M** [comment]
I apologise for inactivity. The new version has been disabled now (hopefully only for the time being).
I guess we're done with this PR now... thanks!

**Comment by Begasus** [comment]
> I apologise for inactivity. The new version has been disabled now (hopefully only for the time being).
> I guess we're done with this PR now... thanks!

Thanks for the change, ready to merge, maybe at one point when acceleration works better it can be enabled as it is :+1:

**Pull request merged**